src/pure/Queue.mo

```diff
@@ -308,16 +372,16 @@ module {
   ///
   /// Example:
   /// ```motoko include=import
-  ///
-  /// Queue.all<Nat>(
-  ///   (?(1, ?(2, ?(3, null))), 3, null),
-  ///   func n = n > 1
-  /// ); // => false
+  /// persistent actor {
+  ///   let queue = Queue.fromIter([1, 2, 3].values());
+  ///   let allGreaterThanOne = Queue.all<Nat>(queue, func n = n > 1);
+  ///   assert not allGreaterThanOne; // false because 1 is not > 1
+  /// }
   /// ```
   ///
   /// Runtime: O(size)
   ///
-  /// Space: O(1)
+  /// Space: O(size) as the current implementation uses `values` to iterate over the queue.
```

Comment:
O(size) but preserves order -- Do we even want more details like that ^ in the docs?

Reply:
It seems reasonable to add more details for any possibly unexpected time or space complexity.